Detectors/PHOS/calib/src/PHOSEnergyCalibDevice.cxx

```diff
@@ -85,8 +100,76 @@ void PHOSEnergyCalibDevice::endOfStream(o2::framework::EndOfStreamContext& ec)
   if (mPostHistos) {
     postHistosCCDB(ec);
   }
+  writeOutFile();
+}
+
+void PHOSEnergyCalibDevice::stop()
+{
+  // DDS stop method: simply write the latest tree
+  LOG(info) << "STOP: writing output digits file";
+  writeOutFile();
+}
+void PHOSEnergyCalibDevice::fillOutputTree()
+{
+
+  if (mOutputDigits.size() == 0) { // nothing to fill
+    return;
+  }
+
+  LOG(info) << "Filling tree with " << mOutputDigits.size() << " digits";
+  if (!mFileOut) { // create file and tree
+    mFileName = mOutputDir + fmt::format("PHOS_CalibDigits_{}.root", mRunNumber);
+    LOG(info) << "Creating new tree for PHOS calib digits, output file=" << mFileName;
+    mFileOut = std::make_unique<TFile>(mFileName.c_str(), "recreate");
+    mTreeOut = std::make_unique<TTree>("phosCalibDig", "O2 PHOS calib tree");
+    mFileMetaData = std::make_unique<o2::dataformats::FileMetaData>();
+  }
+  auto* br = mTreeOut->GetBranch("PHOSCalib");
+  auto* pptr = &mOutputDigits;
+  if (br) {
+    LOG(info) << " Using existing branch";
+    br->SetAddress(&pptr);
+  } else {
+    LOG(info) << " Create new branch";
+    br = mTreeOut->Branch("PHOSCalib", &pptr);
+  }
+  int res = br->Fill();
+  br->ResetAddress();
 }
 
+void PHOSEnergyCalibDevice::writeOutFile()
+{
+  // write collected vector and metadata
+  if (!mTreeOut) { // nothing to write,
+    return;
+  }
+  LOG(info) << "Writing calibration digits";
+  mFileOut->cd();
+  int nbits = mTreeOut->Write();
```

Comment:
you are filling the branch, but you don't set the number of entries in the tree, it will stay 0. Before writing you should do 
```
auto* br = mTreeOut->GetBranch("PHOSCalib");
if (br) {
 mTreeOut->SetEntries(br->GetEntries());
}
```
Alternatively, since you have just 1 branch, you can do simply `mTreeOut->Fill();` instead of `br->Fill();`